chitchat/src/state.rs

```diff
@@ -165,43 +187,43 @@ impl ClusterState {
         // Remove nodes to reset.
         self.node_states
             .retain(|chitchat_id, _| !delta.nodes_to_reset.contains(chitchat_id));
-        // And apply delta.
+
+        // Apply delta.
         for (chitchat_id, node_delta) in delta.node_deltas {
-            let mut node_state_map = self
-                .node_states
-                .entry(chitchat_id)
-                .or_insert_with(NodeState::default);
+            let mut node_state = self.node_states.entry(chitchat_id).or_default();
+
+            if node_state.heartbeat < node_delta.heartbeat {
+                node_state.heartbeat = node_delta.heartbeat;
+                node_state.last_heartbeat = Instant::now();
+            }
+            node_state.max_version = node_state.max_version.max(node_delta.max_version);
 
             for (key, versioned_value) in node_delta.key_values {
-                node_state_map.max_version =
-                    node_state_map.max_version.max(versioned_value.version);
-                let entry = node_state_map.key_values.entry(key);
+                let entry = node_state.key_values.entry(key);
                 match entry {
-                    Entry::Occupied(mut record) => {
-                        if record.get().version >= versioned_value.version {
-                            // Due to the message passing being totally asynchronous, it is not an
-                            // error to receive updates that are already obsolete.
-                            continue;
+                    Entry::Occupied(mut occupied_entry) => {
+                        let local_versioned_value = occupied_entry.get_mut();
+                        // Due to the message passing being totally asynchronous, it is not an
```

Comment:
The comment is a bit weird now as you remove the "continue" :)
I would change it slightly and say something like this

```
Obsolete values are ignored. Due to the message passing being totally asynchronous, it is not a...
```